compiler/src/dmd/cparse.d

```diff
@@ -1823,6 +1823,8 @@ final class CParser(AST) : Parser!AST
                 {
                     if (tt.id || tt.tok == TOK.enum_)
                     {
+                        if (!tt.id && id)
```

Comment:
Does this need to be limited to just enums?

Reply:
It seems that it is enough to limit this to enums. I have just done some tests, and it seems that structs and unions defined in same way works fine in latest released dmd (do not need any fixes), thus there are no changes needed for them.